### PR TITLE
feat(YouTube - Download Playlist Button): Add playlist download button

### PR DESCRIPTION
**Pull request by Francesco146**
ref: https://github.com/anddea/revanced-integrations/pull/26

### CODE REVIEW

**Comment by YT-Advanced** [comment]
### Todo:
- [x] Forced turn on playlist download icon
- [x] Hook download playlist actions
- [ ] Seperate `Hook Download Playlist Button` and `Hook Download Button` from Overlay Buttons

File: src/main/kotlin/app/revanced/patches/youtube/misc/downloadactions/HookDownloadActionsPatch.kt [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+        // endregion
+
+        /**
+         * Add settings
+         */
+        SettingsPatch.addPreference(
+            arrayOf(
+                "PREFERENCE_SCREEN: GENERAL",
+                "SETTINGS: HOOK_PLAYLIST_DOWNLOAD_BUTTON"
```
Comment (Francesco146) [inline src/main/kotlin/app/revanced/patches/youtube/misc/downloadactions/HookDownloadActionsPatch.kt:172]: needs to be updated

Comment (YT-Advanced) [reply src/main/kotlin/app/revanced/patches/youtube/misc/downloadactions/HookDownloadActionsPatch.kt]: Done

**Comment by Francesco146** [comment]
last todo is to add a package name selector so we don't have the dependency of Overlay buttons

File: src/main/resources/youtube/settings/host/values/strings.xml [hunk 2]
```diff
@@ -63,6 +63,9 @@ Tap here to learn more about DeArrow."</string>
     <string name="revanced_bypass_image_region_restrictions_summary_off">Using original image host.\n\nEnabling this can fix missing images that are blocked in some regions.</string>
     <string name="revanced_bypass_image_region_restrictions_summary_on">Using image host yt4.ggpht.com.</string>
     <string name="revanced_bypass_image_region_restrictions_title">Bypass image region restrictions</string>
+    <string name="revanced_hook_playlist_download_button_title">Override playlist download action</string>
+    <string name="revanced_hook_playlist_download_button_summary_off">Playlist download will behave like the original, and button might not appear.</string>
+    <string name="revanced_hook_playlist_download_button_summary_on">Playlist download button will be present, and the download will be handled by YTDLnis.</string>
```
Comment (YT-Advanced) [inline src/main/resources/youtube/settings/host/values/strings.xml:68]: Rename the hook video download button to match with the issues that I have sent u, then move the strings to this positions

File: src/main/resources/youtube/settings/xml/revanced_prefs.xml [hunk 3]
```diff
@@ -106,6 +106,9 @@
 	<!-- SETTINGS: REMOVE_VIEWER_DISCRETION_DIALOG
 	<SwitchPreference android:title="@string/revanced_remove_viewer_discretion_dialog_title" android:key="revanced_remove_viewer_discretion_dialog" android:defaultValue="false" android:summary="@string/revanced_remove_viewer_discretion_dialog_summary" />SETTINGS: REMOVE_VIEWER_DISCRETION_DIALOG -->
 
+	<!-- SETTINGS: HOOK_DOWNLOAD_ACTIONS
+	<SwitchPreference android:title="@string/revanced_hook_playlist_download_button_title" android:key="revanced_hook_playlist_download_button" android:defaultValue="true" android:summaryOn="@string/revanced_hook_playlist_download_button_summary_on" android:summaryOff="@string/revanced_hook_playlist_download_button_summary_off"/>SETTINGS: HOOK_DOWNLOAD_ACTIONS -->
```
Comment (YT-Advanced) [inline src/main/resources/youtube/settings/xml/revanced_prefs.xml:110]: Hmm move the `Override video download button` to this settings

File: src/main/kotlin/app/revanced/patches/youtube/utils/resourceid/SharedResourceIdPatch.kt [hunk 4]
```diff
@@ -118,6 +118,7 @@ object SharedResourceIdPatch : ResourcePatch() {
     var YtOutlineXWhite = -1L
     var YtPremiumWordMarkHeader = -1L
     var YtWordMarkHeader = -1L
+    var AccessibilityOfflineButtonSync = -1L
```
Comment (YT-Advanced) [inline src/main/kotlin/app/revanced/patches/youtube/utils/resourceid/SharedResourceIdPatch.kt:121]: Alphabetical order 😭

**Comment by anddea** [comment]
Thank you for your contribution, guys!

**Comment by Francesco146** [comment]
thankss, was tough but a good!

**Pull request merged**